plugins/module_utils/mso.py

```diff
@@ -490,25 +498,30 @@ def vrf_dict_from_ref(self, data):
         }
 
     def dict_from_ref(self, data):
-        ref_regex = re.compile(r'\/schemas\/(.*)\/templates\/(.*)\/(.*)\/(.*)')
-        dic = ref_regex.search(data)
-        schema_id = dic.group(1)
-        template_name = dic.group(2)
-        category = dic.group(3)
-        name = dic.group(4)
-        uri_map = {
-            'vrfs': ['vrfName', 'schemaId', 'templateName'],
-            'bds': ['bdName', 'schemaId', 'templateName'],
-            'filters': ['filterName', 'schemaId', 'templateName'],
-            'contracts': ['contractName', 'schemaId', 'templateName'],
-            'l3outs': ['l3outName', 'schemaId', 'templateName']
-        }
-        result = {
-            uri_map[category][0]: name,
-            uri_map[category][1]: schema_id,
-            uri_map[category][2]: template_name,
-        }
-        return result
+        if data:
+            ref_regex = re.compile(r'\/schemas\/(.*)\/templates\/(.*)\/(.*)\/(.*)')
+            dic = ref_regex.search(data)
+            if dic:
```

Comment:
if dic and 'group' in dic:

Reply:
Left it as-
If dic is not None:

We can make a change if we hit something down the road.